### PR TITLE
ci(Mergify): configuration update

### PR DESCRIPTION
**Pull request by dave-tucker**
This change has been made by @dave-tucker from Mergify config editor.

### CODE REVIEW

**Comment by dave-tucker** [comment]
This does 3 things:

1. Automatically merges Dependabot PRs that pass CI
2. Labels all PRs 🙏 which was previously done manually for the release note generation
3. Assigns Alessandro to review any PR for which `xtask/public-api/aya.txt` has changed

**Comment by tamird** [comment]
> 

Can you put this in commit message please?

File: .mergify.yml [hunk 1]
```diff
@@ -0,0 +1,101 @@
+pull_request_rules:
+  - name: automatic merge for Dependabot pull request that pass CI
+    conditions:
+      - author=dependabot[bot]
+      - check-success=build-workflow-complete
+    actions:
+      merge:
```
Comment (tamird) [inline .mergify.yml:7]: can we have it leave DP a comment instead?

File: .mergify.yml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+  # REVIEW MANAGEMENT
+
+  - name: ask alessandrod to review public API changes
+    conditions:
+      - files=xtask/public-api/aya.txt
+      - -closed
+    actions:
+      comment:
+        message: "Hey @alessandrod, this pull request changes the Aya API and requires your review."
```
Comment (tamird) [inline .mergify.yml:18]: nit: "changes the public API"

File: .mergify.yml [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+      - files~=^aya/
+      - -closed
+    actions:
+      label:
+        add:
+          - aya
+
+  - name: add labels for aya-bpf PRs
+    conditions:
+      - files~=^(bpf/aya-bpf/|bpf/aya-bpf-cty/|bpf/aya-bpf-bindings/|aya-bpf-macros/)
```
Comment (tamird) [inline .mergify.yml:57]: should this be `(^bpf/|aya-bpf-(bindings|macros)/)`?

File: .mergify.yml [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    actions:
+      label:
+        add:
+          - aya-bpf
+
+  - name: add labels for aya-log PRs
+    conditions:
+      - files~=^(aya-log/|aya-log-common/)
```
Comment (tamird) [inline .mergify.yml:65]: ditto, can simplify to `^aya-log(-common)?/`

File: .mergify.yml [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+  - name: add labels for aya-tool PRs
+    conditions:
+      - files~=^aya-tool/
+    actions:
+      label:
+        add:
+          - aya-tool
+
+  - name: add labels for CI/Test PRs
```
Comment (tamird) [inline .mergify.yml:95]: should this group include xtask and be named appropriately?

**Review by tamird — COMMENTED** [review]
This is neat!

File: .mergify.yml (hunk 1 above)
Comment (dave-tucker) [reply .mergify.yml:7]: DP comment?

Comment (tamird) [reply .mergify.yml:7]: I don't know how smart mergify is, but DP is pretty smart. You can omit the check success thing if you leave a comment, and DP will also rebase the PR (and perhaps regenerate it completely in complex cases e.g. Cargo.lock conflicts).

File: .mergify.yml [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+      - files~=^aya/
+      - -closed
+    actions:
+      label:
+        add:
+          - aya
+
+  - name: add labels for aya-bpf PRs
+    conditions:
+      - files~=^(bpf/aya-bpf(-cty|bindings|macros)?/)
```
Comment (tamird) [inline .mergify.yml:57]: this looks wrong? aya-bpf-macros is at the top level, not in bpf/

Please take a look at my original suggestion.

File: .mergify.yml [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+
+  # REVIEW MANAGEMENT
+
+  - name: ask alessandrod to review public API changes
+    conditions:
+      - files=xtask/public-api/aya.txt
+      - -closed
+    actions:
+      comment:
+        message: "Hey @alessandrod, this pull request changes the Aya Public API and requires your review."
```
Comment (tamird) [inline .mergify.yml:18]: my point here was to drop "Aya", but I see now that this is only applied to aya, and not any of the other crates, so maybe it's fine.

File: .mergify.yml [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+    conditions:
+      - files~=^aya-tool/
+    actions:
+      label:
+        add:
+          - aya-tool
+
+  - name: add labels for CI/Test and Automation PRs
+    conditions:
+      - files~=^(.github/|test/|xtask/)
```
Comment (tamird) [inline .mergify.yml:97]: slash can be on the outside

File: .mergify.yml [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+    conditions:
+      - files~=^aya-log(-common)?/
+    actions:
+      label:
+        add:
+          - aya-log
+
+  - name: add labels for aya-log-ebpf PRs
+    conditions:
+      - files~=^bpf/aya-log-(ebpf|common)/
```
Comment (tamird) [inline .mergify.yml:73]: this is wrong; aya-log-common is at the top level

File: .mergify.yml [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+      - -conflict
+    actions:
+      label:
+        remove:
+          - needs-rebase
+
+  - name: add labels for aya PRs
+    conditions:
+      - files~=^aya/
+      - -closed
```
Comment (tamird) [inline .mergify.yml:49]: how come only aya gets this filter?

File: .mergify.yml (hunk 6 above)
Comment (dave-tucker) [reply .mergify.yml]: This was your suggestion:

> should this be `(^bpf/|aya-bpf-(bindings|macros)/)`?

That would catch `bpf/aya-log-ebpf` also + aya-bpf-bindings are in the bpf directory, not top-level.
I've now gone with something that should catch `aya-bpf-macros` + only the aya-bpf related dirs in `bpf`.

File: .mergify.yml [hunk 11]
```diff
@@ -0,0 +1,100 @@
+pull_request_rules:
+  - name: automatic merge for Dependabot pull request that pass CI
+    conditions:
+      - author=dependabot[bot]
+      - check-success=build-workflow-complete
```
Comment (tamird) [inline .mergify.yml:5]: this can be removed; DP won't merge unless all checks pass

File: .mergify.yml (hunk 6 above)
Comment (tamird) [reply .mergify.yml]: Thanks, my suggestion was incorrect.

**Review by tamird — APPROVED** [review]

File: .mergify.yml (hunk 11 above)
Comment (dave-tucker) [reply .mergify.yml]: 👍 removed.

**Comment by dave-tucker** [comment]
Will merge this on green (unless someone else gets to it first) ✅

**Pull request merged**